### PR TITLE
Feat/context upgrades

### PR DESCRIPTION
**Pull request by KillianCourvoisier**
This PR completes the missing login/logout scenarios of previous branch `feat/enhancedFetchContracts` and adds context upgrades on the konnector.

We're now logging out on every execution, at the beginning if needed (to avoid in between scenarios where you could be sort of "half logged" and mismatch sessions between Orange and Sosh konnectors executions) bringing us to 4 scenarios to handle.

- First execution (no account, not connected) : Let user login, intercept credentials, fetch and logout
- AutoLogin (account and not connected) : autoLogin, fetch and logout
- AutoLogin (account and already connected because last execution crashed) : preventive logout, autoLogin, fetch, logout
- Bonus case ( konnector crashed on last execution, and somehow user remove cozy account => No account, already connected) : preventive logout, let user login, intercept credentials, fetch and logout

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -389,6 +402,8 @@ class SoshContentScript extends ContentScript {
     await this.navigateToPersonalInfos()
     await this.runInWorker('getIdentity')
     await this.saveIdentity(this.store.infosIdentity)
+    // Logging out every run to avoid in between scenarios and sosh/orange mismatched sessions
+    await this.logout()
```
Comment (doubleface) [inline src/index.js:474]: If any error is thrown, this.logout won't be called. If you want to be sure that the logout function is called, you should call it in the beginning of the execution.

File: src/index.js [hunk 2]
```diff
@@ -100,24 +100,22 @@ class SoshContentScript extends ContentScript {
...
-        ),
-        this.waitForElementInWorker('div[class*="captcha_responseContainer"]'),
-        this.waitForElementInWorker('#undefined-label'),
-        this.waitForElementInWorker('#oecs__popin-icon-Identification'),
-        this.waitForElementInWorker('#changeAccountLink'),
-        this.waitForErrorUrl()
-      ],
-      'navigateToLoginForm: waiting for login page load'
-    )
+    await Promise.race([
```
Comment (doubleface) [inline src/index.js:110]: Why did you remove the use of PromiseRaceWithError ?

File: src/index.js [hunk 3]
```diff
@@ -242,8 +206,32 @@ class SoshContentScript extends ContentScript {
 
   async ensureNotAuthenticated() {
     this.log('info', '🤖 ensureNotAuthenticated starts')
-    await this.goto(LOGOUT_URL)
-    await this.waitForElementInWorker('#oecs__connexion')
+    await this.goto(DEFAULT_PAGE_URL)
+    await this.waitForElementInWorker(
+      '#login, #password, div[class*="captcha_responseContainer"], button[data-testid="button-keepconnected"], #oecs__connexion, #oecs__connecte-se-deconnecter'
+    )
+    await this.handleCaptcha()
```
Comment (doubleface) [inline src/index.js:212]: This is weird to have the handle of captcha in ensureNotAuthenticated since we did not try any authentication yet.

File: src/index.js (hunk 2 above)
Comment (KillianCourvoisier) [reply src/index.js:110]: It was unvoluntarily, will be fixed 👍

File: src/index.js (hunk 3 above)
Comment (KillianCourvoisier) [reply src/index.js:212]: I know but it happens during the development, so at least we're covering the case 🤷

File: src/index.js (hunk 1 above)
Comment (KillianCourvoisier) [reply src/index.js:474]: Yep, it is handle in the navigateToLoginForm method, at the start of the execution to cover this case. That way we just have to handle 3 scenarios, thefirst execution, the autoLogin, and the logout + autoLogin for the most part to avoid orange/sosh mixing sessions 👍

File: src/index.js [hunk 4]
```diff
@@ -37,6 +37,60 @@ class SoshContentScript extends ContentScript {
...
+      if (login && password) {
+        this.store.userCredentials = { login, password }
+      } else {
+        this.log('warn', 'Did not manage to intercept credentials')
+      }
+    }
+  }
+
+  async onWorkerReady() {
+    function addClickListener() {
```
[the remark this answers is not in the document]
Comment (KillianCourvoisier) [reply src/index.js:51]: As we're using the `waitForElementNoReload` worker's method, we'll only set the listener once the form inputs are detected, so i think this is not mandatory here. As we already have issues with waiting for the DOM in other konnectors, maybe this is the best way to ensure the listener is sat only when needed, and not everytime the worker has sent the DOMContentLoaded event, but maybe i'm wrong. What you think ?

File: src/index.js (hunk 3 above)
Comment (doubleface) [reply src/index.js:212]: Ok, let's add a comment to know the reason for this then

File: src/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (doubleface) [reply src/index.js:51]: Ok, I did not notice the use of waitForElementNoReload :+1:

**Review by doubleface — APPROVED** [review]
